app/src/components/nav/AppShell.tsx

```diff
@@ -122,7 +122,7 @@ const NavSidebar = () => {
               as={ChakraLink}
               justifyContent="start"
               onClick={() => {
-                signIn("github").catch(console.error);
```

Comment:
How is the behavior in prod affected if we call `signIn` without passing "github" explicitly?

Reply:
The `signIn()` function without arguments defaults to redirecting the user to the NextAuth.js sign-in page, where they can select their preferred sign-in method from the configured options.

Documentation : https://next-auth.js.org/getting-started/client#signin

This is how it would look on prod. Currently on prod we directly go to the GitHub login page.

<img width="566" alt="image" src="https://github.com/OpenPipe/OpenPipe/assets/35342595/bcbcdfda-0e57-4994-9fdb-0f3c49d50a47">

Reply:
Can this PR be written in such a way that it doesn't affect prod behavior?

Reply:
I'm a little confused by the updated behavior. Is there any reason to look for the literal "your_client_id", instead of making `GITHUB_CLIENT_ID` an optional field and checking for its presence?

Reply:
Yes! Right now in the  `.env.example` file we ask users to copy,  has a default value for GITHUB environment variables. Hence the literal comparison. However If we're to just check for presence of the field we will have to update the env file as such.

```
# Before
GITHUB_CLIENT_ID="your_client_id"
GITHUB_CLIENT_SECRET="your_secret"

# After
GITHUB_CLIENT_ID=""
GITHUB_CLIENT_SECRET=""
```

I am fine with either of these approaches and I agree with you that checking for presence might be slightly better.

Reply:
Let's remove the variables from the example file entirely, mark them as optional in env.mjs, and then just check if they're set or not to choose which branch to take. Thanks!

Reply:
Sure! I have made the changes. Seems a lot cleaner now.